apps/site/src/pages/settings/billing/upgrade/index.page.tsx

```diff
@@ -0,0 +1,781 @@
+        return "hobby";
+      } else if (currentSubscriptionTier === "hobby") {
+        return "pro";
+      }
+    }
+  }, [router, currentSubscriptionTier]);
+
+  const fetchSubscription = useCallback(async () => {
+    const { data } = await internalApi.getSubscription({});
+
+    setSubscription(data.subscription);
+  }, []);
+
+  const isUpgradingExistingPaidSubscription = useMemo<boolean>(
+    () =>
+      !!upgradedSubscriptionTier &&
+      isPaidSubscriptionTier(currentSubscriptionTier) &&
+      currentSubscriptionTier !== upgradedSubscriptionTier,
+    [currentSubscriptionTier, upgradedSubscriptionTier],
+  );
+
+  useEffect(() => {
+    if (isUpgradingExistingPaidSubscription) {
+      void fetchSubscription();
+    }
+  }, [isUpgradingExistingPaidSubscription, fetchSubscription]);
+
+  const stripeElementsOptions = useMemo<StripeElementsOptions>(
+    () => ({
+      clientSecret,
+    }),
+    [clientSecret],
+  );
+
+  const createSubscription = async (params: { tier: PaidSubscriptionTier }) => {
+    const { data } = await internalApi.createSubscription({
+      subscriptionTier: params.tier,
+    });
+
+    setSubscriptionId(data.subscriptionId);
+    setClientSecret(data.clientSecret);
+  };
+
+  useEffect(() => {
+    if (upgradedSubscriptionTier && !subscriptionId) {
+      if (currentSubscriptionTier === "free") {
+        void createSubscription({ tier: upgradedSubscriptionTier });
+      }
+    }
+  }, [currentSubscriptionTier, upgradedSubscriptionTier, subscriptionId]);
+
+  const handleUpgradedSubscription = useCallback(() => {
+    /**
+     * The BP user in the database is updated by the stripe webhook in the
+     * internal API. To reflect the updated state of the user before this
+     * db write has taken place, we can manually update the user object
+     * client-side with the anticipated values for `stripeSubscriptionStatus`
+     * and `stripeSubscriptionTier`.
+     */
+    setUser((prevUser) => {
+      if (typeof prevUser !== "object") {
+        return prevUser;
+      }
+
+      return {
+        ...prevUser,
+        stripeSubscriptionStatus: "active",
+        stripeSubscriptionTier: upgradedSubscriptionTier,
+      };
+    });
+
+    void router.push("/settings/billing");
+  }, [setUser, router, upgradedSubscriptionTier]);
+
+  const upgradeSubscription = async (params: {
+    tier: Exclude<PaidSubscriptionTier, "hobby">;
+  }) => {
+    setIsUpgradingSubscription(true);
+
+    await internalApi.updateSubscription({
+      updatedSubscriptionTier: params.tier,
+    });
+
+    setIsUpgradingSubscription(false);
+
+    /** @todo: catch payment errors */
+
+    handleUpgradedSubscription();
+  };
+
+  const _currentPeriodEnd = useMemo(() => {
+    if (!subscription || !subscription.current_period_end) {
+      return undefined;
+    }
+
+    const date = getDateFromStripeDate(subscription.current_period_end);
+
+    return `${date.getDate()}/${date.getMonth() + 1}/${date.getFullYear()}`;
```

Comment:
Can't we use the util `dateToHumanReadable` here?

Reply:
Yes, but we don't actually need this variable as I had originally thought. Have removed it in https://github.com/blockprotocol/blockprotocol/pull/901/commits/bd1bb4e00207c458c3ec123544d48a6a978a8d04, and the related `subscription` variable in https://github.com/blockprotocol/blockprotocol/pull/901/commits/dc5d67fbc9c413a16f440847f27e245bd37c4893